language/tools/move-mv-llvm-compiler/utils/testgen/src/main.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) The Diem Core Contributors
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#[macro_use]
+extern crate serde_derive;
+extern crate mustache;
+
+use std::{collections::HashMap, fs::File};
+
+static TESTCASE_TEMPLATE: &str = r##"// {{result}}
+module 0x101::Test1 {
+  public fun test_{{name}}({{parameters}}): {{return_type}} {
+    {{body}}
+  }
+}
+
+script {
+  fun main() {
+    assert!(0x101::Test1::test_{{name}}({{valid_arguments}}) == {{expected}}, 10);  // Ok: source fits in dest.
+
+    0x101::Test1::test_{{name}}({{wrong_arguments}});  // Abort: source too big.
+  }
+}
+"##;
+
+#[derive(Serialize)]
+struct Instance {
+    name: String,
+    parameters: String,
+    return_type: String,
+    body: String,
+    result: String,
+    valid_arguments: String,
+    wrong_arguments: String,
+    expected: String,
+}
+
+fn generate_cast_tests() {
+    // First the string needs to be compiled.
+    let template = mustache::compile_str(TESTCASE_TEMPLATE).unwrap();
+
```

Comment:
This program is similar to what I did to generate the original `cast*` test cases with a one-off python repl session. But this will be extensible to other scenarios. We have many integer operations without good runnable test coverage.

Also, I noticed that the Move Prover seems to have a built-in mutation (fuzzer) testing mechanism, but I have not studied it. That may be something we could use or modify for our use (`language/move-prover/mutation`).

Reply:
I used the cast* tests as a starting point to define a template. This template can be either generalized further or other templates could be added for another kind of test cases.

Reply:
Very nice, LGTM.
As an aside, auto-generating some struct cases in the future would be good to. Those are tougher because of variable number of fields, nested structure members, etc.